pkg/component/component.go

```diff
@@ -0,0 +1,229 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package component
+
+import (
+	"fmt"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+)
+
+// Unit is a single unit that needs to be running inside a component.
+type Unit struct {
+	ID     string
+	Type   client.UnitType
+	Config map[string]interface{}
+}
+
+// Component is a set of units that needs to run.
+type Component struct {
+	Spec  InputRuntimeSpec
+	Units []Unit
+}
+
+// ToComponents returns the components that should be running based on the policy and the current runtime specification.
+func (r *RuntimeSpecs) ToComponents(policy map[string]interface{}) ([]Component, error) {
+	outputsMap, err := toIntermediate(policy)
+	if err != nil {
+		return nil, err
+	}
+	if outputsMap == nil {
+		return nil, nil
+	}
+	var components []Component
+	for outputName, output := range outputsMap {
+		if !output.enabled {
+			// skip; not enabled
+			continue
+		}
+
+		// merge aliases into same input type
+		inputsMap := make(map[string][]inputI)
+		for inputType, inputs := range output.inputs {
+			realInputType, ok := r.aliasMapping[inputType]
+			if ok {
+				inputsMap[realInputType] = append(inputsMap[realInputType], inputs...)
+			} else {
+				inputsMap[inputType] = append(inputsMap[inputType], inputs...)
+			}
+		}
+
+		for inputType, inputs := range inputsMap {
+			inputSpec, err := r.GetInput(inputType)
+			if errors.Is(err, ErrNotSupportedOnPlatform) {
+				return nil, fmt.Errorf("invalid 'input.%d' type '%s' is not supported on this platform", inputs[0].idx, inputType)
+			} else if err != nil {
+				return nil, fmt.Errorf("invalid 'input.%d' type '%s' is an unknown input", inputs[0].idx, inputType)
+			}
+			if !containsStr(inputSpec.Spec.Outputs, output.outputType) {
+				return nil, fmt.Errorf("invalid 'input.%d' type '%s' does not support output type '%s'", inputs[0].idx, inputType, output.outputType)
+			}
+			units := make([]Unit, 0, len(inputs)+1)
+			for _, input := range inputs {
+				if !input.enabled {
+					// skip; not enabled
+					continue
+				}
+				units = append(units, Unit{
+					ID:     fmt.Sprintf("%s-%s-%s", inputSpec.InputType, outputName, input.id),
+					Type:   client.UnitTypeInput,
+					Config: input.input,
+				})
+			}
+			if len(units) > 0 {
+				units = append(units, Unit{
+					ID:     fmt.Sprintf("%s-%s", inputSpec.InputType, outputName),
```

Comment:
Every unit must have an ID in the V2 protocol.

Reply:
I didn't explain myself correctly, I was asking if we need fleet to send us a `output.id` but we don't because the `outputName `is unique per configuration.

Reply:
Ah. Yes `outputName` is unique so we use that as the key.